File: src/model/Produto.js
const { Sequelize } = require("sequelize");
const sequelize = require("../utils/db");
const { Usuario } = require("./Usuarios");

const Produto = sequelize.define(
  "Produto",
  {
    produto_id: {
      type: Sequelize.INTEGER,
      primaryKey: true,
      autoIncrement: true,
    },
    empresa_id: {
      type: Sequelize.INTEGER,
      allowNull: false,
      references: {
        model: "usuarios",
        key: "usuario_id",
      },
    },
    nome: {
      type: Sequelize.STRING,
      allowNull: false,
    },
    valor: {
      type: Sequelize.FLOAT,
      allowNull: false,
    },
    valor_custo: {
      type: Sequelize.FLOAT,
      allowNull: false,
    },
    quantidade: {
      type: Sequelize.INTEGER,
      allowNull: false,
    },
    tipo_comercializacao: {
      type: Sequelize.STRING,
      allowNull: false,
    },
    tipo_produto: {
      type: Sequelize.STRING,
      allowNull: false,
    },
    empresas_autorizadas: {
      type: Sequelize.ARRAY(Sequelize.INTEGER),
      allowNull: true,
      defaultValue: [],
    },
    foto_principal: {
      type: Sequelize.TEXT,
      allowNull: true,
    },
    data_cadastro: {
      type: Sequelize.DATE,
      defaultValue: Sequelize.literal("CURRENT_TIMESTAMP"),
    },
    data_update: {
      type: Sequelize.DATE,
      defaultValue: Sequelize.literal("CURRENT_TIMESTAMP"),
    },
  },
  {
    schema: "public",
    tableName: "produtos",
    timestamps: false,
  }
);

Produto.belongsTo(Usuario, { foreignKey: "empresa_id", as: "Empresa" });

module.exports = { Produto };

File: src/main/java/com/elo/api/models/Produto.java
package com.elo.api.models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "produtos", schema = "public")
@Data
@NoArgsConstructor
public class Produto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "produto_id")
    private Integer produtoId;

    @ManyToOne
    @JoinColumn(name = "empresa_id", nullable = false)
    private Usuario empresa;

    @Column(nullable = false)
    private String nome;

    private BigDecimal valor;

    @Column(name = "valor_custo")
    private BigDecimal valorCusto;

    private Integer quantidade;

    @Column(name = "tipo_comercializacao")
    private String tipoComercializacao;

    @Column(name = "tipo_produto")
    private String tipoProduto;

    @ElementCollection
    @CollectionTable(name = "produto_empresas_autorizadas",
            joinColumns = @JoinColumn(name = "produto_id"))
    @Column(name = "empresa_id")
    private List<Integer> empresasAutorizadas = new ArrayList<>();

    @Column(name = "foto_principal")
    private String fotoPrincipal;

    @OneToMany(mappedBy = "produto", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<FotoProduto> fotos = new ArrayList<>();

    @Column(name = "data_cadastro")
    private LocalDateTime dataCadastro = LocalDateTime.now();

    @Column(name = "data_update")
    private LocalDateTime dataUpdate;
}
